### ISSUE TITLE
Data upload messes up timestamps

### ISSUE THREAD

**Issue by Acefall** [data-hydenv/hydenv-database#29]
After running `python -m hydenv examples hobo --terms=[WT21]` the timestamps in the `raw_data` table are messed up. It confused 2020-12-11 (yyyy-mm-dd) to 2020-11-12 (yyyy-mm-dd). Is this intended and we have to modify the timestamp for our analysis? If not, this is very confusing.
Attached is a screenshot from pgAdmin:
![timestamps](https://user-images.githubusercontent.com/76061447/104909314-f45b1900-5987-11eb-9301-85cf39459b21.png)

**Comment by mmaelicke** [comment]
This is definitely a bug. Have to check this, if there is a difference between last years raw data and this year's raw data. Then another handler has to be added to the upload script.
Do you know if this happens for all data this year or just for the shown `device_id`?

Thank you for noticing and reporting.

**Comment by Acefall** [comment]
I checked for two different devices from WT21.
How do I proceeed? Create a view that fixes the timestamp?

**Comment by mmaelicke** [comment]
I imagine this could become quite complex. This does only affect the `raw_data` table, right? Then, for proceeding with the SQL exercise for the report, the `data` table is used anyway.

I will either come up with an update in the CLI, or a fixing SQL command, I first have to check where exactly things are going wrong.

**Comment by Acefall** [comment]
I just realized that we only need the `data` table and not the `raw_data` table. Thus this is no longer an issue for me.

**Comment by mmaelicke** [comment]
Yeah, that's actually the problem. The date format changed: 

`cat` from **2020**:
```
"Plot Title: 10349994"
"#","Date Time, GMT+01:00","Temp, °C (LGR S/N: 10349994, SEN S/N: 10349994)","Intensity, Lux (LGR S/N: 10349994, SEN S/N: 10349994)","Host Connected (LGR S/N: 10349994)","Coupler Attached (LGR S/N: 10349994)","Stopped (LGR S/N: 10349994)","End Of File (LGR S/N: 10349994)"
1,12/14/2019 12:00:00 AM,5.655,0.0,,,,
2,12/14/2019 12:10:00 AM,5.757,0.0,,,,
3,12/14/2019 12:20:00 AM,5.655,0.0,,,,
4,12/14/2019 12:30:00 AM,5.552,0.0,,,,
5,12/14/2019 12:40:00 AM,5.860,0.0,,,,
6,12/14/2019 12:50:00 AM,5.860,0.0,,,,
7,12/14/2019 01:00:00 AM,6.064,0.0,,,,
8,12/14/2019 01:10:00 AM,6.268,0.0,,,,
```

`cat` from **2021**:

```
"Plot Title: 10350087"
"#","Date Time, GMT+01:00","Temp, °C (LGR S/N: 10350087, SEN S/N: 10350087)","Intensity, Lux (LGR S/N: 10350087, SEN S/N: 10350087)","Bad Battery (LGR S/N: 10350087)","Coupler Attached (LGR S/N: 10350087)","Host Connected (LGR S/N: 10350087)","Stopped (LGR S/N: 10350087)","End Of File (LGR S/N: 10350087)"
1,29/11/2020 18:00:00,3.998,0.0,Logged,,,,
2,29/11/2020 18:10:00,3.893,0.0,,,,,
3,29/11/2020 18:20:00,3.788,0.0,,,,,
4,29/11/2020 18:30:00,3.788,0.0,,,,,
5,29/11/2020 18:40:00,3.788,0.0,,,,,
6,29/11/2020 18:50:00,3.788,0.0,,,,,
7,29/11/2020 19:00:00,3.472,0.0,,,,,
8,29/11/2020 19:10:00,3.472,0.0,,,,,
```

Now, I have to check, whether this happened systematically...   @modche, this is kind weird, as the HOBO software created these data dumps, right?

**Comment by mmaelicke** [comment]
At least this happened systematically... Puhh, Heartrate went down from 200bpm again.

### 2020:

```SQL 
select distinct date(tstamp) from raw_data r
join metadata m on r.meta_id=m.id
where term_id=9
order by date(tstamp) ASC
```

Delivers all dates from 14.Dec 2019 to 8. Jan 2020.  So the dates are fine in every single case.

### 2021

```SQL
select distinct date(tstamp) from raw_data r
join metadata m on r.meta_id=m.id
where term_id=11
order by date(tstamp) ASC
```
delivers the 12. of Jan, Feb, Mar ... Nov. 2020, and then 30.11.2020 to 21.12.2020 in the correct order. That means the problem is exactly like @Acefall  described and caused by a different date format in the onset HOBO raw data files. At least it was screwed up in all cases.

Unfortunately, it gets quite complicated to catch all cases here. I'll come up with a strategy...

**Comment by mmaelicke** [comment]
This is now fixed with version `0.9.2`.
However, the CLI now checks explicitly against winter term 2021. If this changed datetime format is also there for next year, we need to create a *master time series* and check if the data was loaded correctly into memory before pushing it into the database.
